diff --git a/src/Plugin/migrate/source/d6/D6Webform.php b/src/Plugin/migrate/source/d6/D6Webform.php
index 2605818..58bd0d0 100644
--- a/src/Plugin/migrate/source/d6/D6Webform.php
+++ b/src/Plugin/migrate/source/d6/D6Webform.php
@@ -281,7 +281,8 @@ class D6Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
         $element['value'] = $this->replaceTokens($element['value']);
       }
 
-      $markup = $indent . $element['form_key'] . ":\n";
+      $element_form_key = strtolower($element['form_key']);
+      $markup = $indent . $element_form_key . ":\n";
       switch ($element['type']) {
         case 'fieldset':
           if ($multiPage && empty($current_page_title)) {
@@ -406,7 +407,7 @@ class D6Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
 
         case 'pagebreak':
           $output = str_replace('{' . $current_page . '_title}', $current_page_title, $output);
-          $current_page = $element['form_key'];
+          $current_page = $element_form_key;
           $markup .= "$indent  '#type': webform_wizard_page\n  '#open': true\n  '#title': {" . $current_page . "_title}\n";
           $current_page_title = t('Page') . ' ' . $pageCnt++;
           break;
diff --git a/src/Plugin/migrate/source/d7/D7Webform.php b/src/Plugin/migrate/source/d7/D7Webform.php
index 4db9bb4..0fb3657 100644
--- a/src/Plugin/migrate/source/d7/D7Webform.php
+++ b/src/Plugin/migrate/source/d7/D7Webform.php
@@ -322,7 +322,8 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
         $element['value'] = $this->replaceTokens($element['value']);
       }
 
-      $markup = $indent . $element['form_key'] . ":\n";
+      $element_form_key = strtolower($element['form_key']);
+      $markup = $indent . $element_form_key . ":\n";
       switch ($element['type']) {
         case 'fieldset':
           if ($multiPage && empty($current_page_title)) {
@@ -452,7 +453,7 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
 
         case 'pagebreak':
           $output = str_replace('{' . $current_page . '_title}', $current_page_title, $output);
-          $current_page = $element['form_key'];
+          $current_page = $element_form_key;
           $markup .= "$indent  '#type': webform_wizard_page\n  '#open': true\n  '#title': {" . $current_page . "_title}\n";
           $current_page_title = t('Page') . ' ' . $pageCnt++;
           break;
@@ -622,7 +623,7 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
         }
 
         if (!$unsupported_condition) {
-          $states[$element_state][] = [':input[name="' . $depedent['form_key'] . '"]' => $element_condition];
+          $states[$element_state][] = [':input[name="' . strtolower($depedent['form_key']) . '"]' => $element_condition];
         }
 
       }
