diff --git a/src/Plugin/migrate/source/d7/D7Webform.php b/src/Plugin/migrate/source/d7/D7Webform.php
index a82d417..7038c53 100644
--- a/src/Plugin/migrate/source/d7/D7Webform.php
+++ b/src/Plugin/migrate/source/d7/D7Webform.php
@@ -582,8 +582,7 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
             break;
 
           case 'set':
-            // Nothing found in D8 :(.
-            $unsupported_condition = TRUE;
+            $element_state = 'set|' . str_replace(' ', '&nbsp;', htmlentities($condition['argument']));
             break;
         }
         // Condition states.
@@ -591,11 +590,14 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
         $depedent = $elements[$condition['source']];
         $depedent_extra = unserialize($depedent['extra']);
         $depedent_extra['items'] = explode("\n", $depedent_extra['items']);
-
+        // Checkboxes conditions handled by definition of checked value
+        // inside state key. This flag will help detenmine checkboxes element.
+        $is_checkboxes = $depedent['type'] == 'select' && empty($depedent_extra['aslist']) && $depedent_extra['multiple'];
         switch ($condition['operator']) {
           case 'equal':
             $element_condition = ['value' => $operator_value];
-            if ($depedent['type'] == 'select' && !$depedent_extra['aslist'] && $depedent_extra['multiple']) {
+            // Specially handle the checkboxes.
+            if ($is_checkboxes) {
               $element_condition = ['checked' => TRUE];
             }
             break;
@@ -605,7 +607,7 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
             $element_state = $condition['invert'] ? 'visible' : 'invisible';
             $element_condition = ['value' => $operator_value];
             // Specially handle the checkboxes.
-            if ($depedent['type'] == 'select' && !$depedent_extra['aslist'] && $depedent_extra['multiple']) {
+            if ($is_checkboxes) {
               $element_condition = ['checked' => TRUE];
             }
 
@@ -638,15 +640,14 @@ class D7Webform extends DrupalSqlBase implements ImportAwareInterface, RollbackA
             break;
         }
 
-        if (!$depedent_extra['aslist'] && $depedent_extra['multiple'] && is_array($depedent_extra['items']) && count($depedent_extra['items']) > 1) {
-          $depedent['form_key'] = $depedent['form_key'] . "[$operator_value]";
+        $dependent_form_key = strtolower($depedent['form_key']);
+        if (!empty($operator_value) && $is_checkboxes) {
+          // Checkboxes conditions handled by definition of checked value
+          // inside state key. Adding value as part state key.
+          $dependent_form_key = $dependent_form_key . "[$operator_value]";
         }
-        elseif (!$depedent_extra['aslist'] && !$depedent_extra['multiple'] && is_array($depedent_extra['items']) && count($depedent_extra['items']) == 1) {
-          $depedent['form_key'] = $depedent['form_key'] . "[$operator_value]";
-        }
-
         if (!$unsupported_condition) {
-          $states[$element_state][] = [':input[name="' . strtolower($depedent['form_key']) . '"]' => $element_condition];
+          $states[$element_state][] = [':input[name="' . $dependent_form_key . '"]' => $element_condition];
         }
 
       }
